centreon/www/include/configuration/configObject/hostgroup/DB-Func.php

```diff
@@ -265,180 +265,631 @@ function multipleHostGroupInDB($hostGroups = array(), $nbrDup = array())
+            );
+
+            // No dataset_filter of type hostgroup found. Create a new one
+            if ($hostGroupDatasetFilters === []) {
+                // get the dataset with the highest ID (last one added) which is the first element of the datasets array
+                $lastDatasetAdded = $datasets[0];
+
+                preg_match('/dataset_for_rule_\d+_(\d+)/', $lastDatasetAdded['dataset_name'], $matches);
+
+                // calculate the new dataset_name
+                $newDatasetName = 'dataset_for_rule_' . $ruleId . '_' . (int) $matches[1] + 1;
+
+                $datasetId = createNewDataset(datasetName: $newDatasetName);
+                linkDatasetToRule(datasetId: $datasetId, ruleId: $ruleId);
+                linkHostGroupToDataset(datasetId: $datasetId, hostGroupId: $hostGroupId);
+                createNewDatasetFilter(datasetId: $datasetId, ruleId: $ruleId, hostGroupId: $hostGroupId);
+            } else {
+                linkHostGroupToDataset(datasetId: $hostGroupDatasetFilters[0]['dataset_id'], hostGroupId: $hostGroupId);
+                // Expend the existing hostgroup dataset_filter
+                $expendedResourceIds = $hostGroupDatasetFilters[0]['dataset_filter_resources'] . ', ' . $hostGroupId;
+
+                updateDatasetFiltersResourceIds(
+                    datasetFilterId: $hostGroupDatasetFilters[0]['dataset_filter_id'],
+                    resourceIds: $expendedResourceIds
+                );
             }
-            unset($resource_list);
         }
+    } else {
+        if (! $centreon->user->admin) {
+            $userResourceAccesses = $centreon->user->access->getResourceGroups();
+            if ($userResourceAccesses !== []) {
+                $statement = $pearDB->prepare(<<<SQL
+                    INSERT INTO `acl_resources_hg_relations` (acl_res_id, hg_hg_id)
+                    VALUES (:aclResourceId, :hostGroupId)
+                    SQL
+                );
+                foreach ($userResourceAccesses as $resourceAccessId => $resourceAccessName) {
+                    $statement->bindValue(':aclResourceId', (int) $resourceAccessId, \PDO::PARAM_INT);
+                    $statement->bindValue(':hostGroupId', (int) $hostGroupId, \PDO::PARAM_INT);
+                    $statement->execute();
+                }
+                unset($userResourceAccesses);
+            }
+        }
+    }
+}
+
+/**
+ * @param int $datasetFilterId
+ * @param string $resourceIds
+ */
+function updateDatasetFiltersResourceIds(int $datasetFilterId, string $resourceIds): void
+{
+    global $pearDB;
+
+    $request = <<<'SQL'
+        UPDATE dataset_filters SET resource_ids = :resourceIds WHERE `id` = :datasetFilterId
+    SQL;
+
+    $statement = $pearDB->prepare($request);
+    $statement->bindValue(':datasetFilterId', $datasetFilterId, \PDO::PARAM_INT);
+    $statement->bindValue(':resourceIds', $resourceIds, \PDO::PARAM_STR);
```

Comment:
The resources_ids column is limited to 255 characters. Are you sure you won't exceed this limit?

Reply:
Yes, since MySQL 5.5, VARCHAR can go up to 65 535 bytes, same size as TEXT... We should be good 👍🏻